### PR TITLE
Periodic and bulk payments support

### PR DESCRIPTION
**Pull request by bdusauso**

### CODE REVIEW

File: lib/ibanity/util.rb [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
-      attributes.keys.reduce({}) do |result, key|
-        result[underscore(key)] = attributes[key]
-        result
+    def self.underscorize(obj)
+      case obj
+      when Array
+        obj.map { |e| underscorize(e) }
+      when Hash
+        obj.keys.reduce({}) do |result, key|
+          result[underscore(key)] = underscorize(obj[key])
```
Comment (cleclefibanity) [inline lib/ibanity/util.rb:9]: underscorize value?

File: lib/ibanity/api/base_resource.rb [hunk 2]
```diff
@@ -92,7 +92,7 @@ def prepare_attributes(raw)
       attributes           = raw["attributes"]    || {}
       meta                 = raw["meta"]          || {}
       params               = base.merge(attributes).merge(meta)
-      Ibanity::Util.underscorize_hash(params)
+      Ibanity::Util.underscorize(params)
```
Comment (cleclefibanity) [inline lib/ibanity/api/base_resource.rb:95]: why underscorize? Shouldn't it be camelized?

Comment (cleclefibanity) [inline lib/ibanity/api/base_resource.rb:95]: Does the output message of the lib output as snake_case to the API ?

Comment (bdusauso) [reply lib/ibanity/api/base_resource.rb:95]: As far as I know the payload is sent in snake case.
It seems the server transforms all payloads in snake case as well, so it doesn't matter if it's camel case or snake case.

Comment (cleclefibanity) [reply lib/ibanity/api/base_resource.rb:95]: ok. No change required then.

File: lib/ibanity/util.rb (hunk 1 above)
Comment (cmw0513) [inline lib/ibanity/util.rb:9]: This is a good improvement, however it is a breaking change... We should definitely add it to the changelog and also bump the major version.

Comment (bdusauso) [reply lib/ibanity/util.rb:9]: I agree

**Pull request merged**